Tests/NIOTransportServicesTests/NIOTSBootstrapTests.swift

```diff
@@ -371,6 +371,60 @@ final class NIOTSBootstrapTests: XCTestCase {
         XCTAssertEqual(try listenerChannel.getOption(NIOTSChannelOptions.multipathServiceType).wait(), .handover)
         XCTAssertEqual(try connectionChannel.getOption(NIOTSChannelOptions.multipathServiceType).wait(), .handover)
     }
+
+    func testNWParametersConfigurator() async throws {
+        final class WaitForConnectionHandler: ChannelInboundHandler, Sendable {
+            typealias InboundIn = Never
+
+            let connectionPromise: EventLoopPromise<Void>
+
+            init(connectionPromise: EventLoopPromise<Void>) {
+                self.connectionPromise = connectionPromise
+            }
+
+            func channelActive(context: ChannelHandlerContext) {
+                self.connectionPromise.succeed()
+            }
+        }
+
+        let group = NIOTSEventLoopGroup(loopCount: 1)
+
+        let configuratorListenerCounter = NIOLockedValueBox(0)
+        let configuratorConnectionCounter = NIOLockedValueBox(0)
+        let waitForConnectionHandler = WaitForConnectionHandler(
+            connectionPromise: group.next().makePromise()
+        )
+
+        let listenerChannel = try await NIOTSListenerBootstrap(group: group)
+            .childChannelInitializer { connectionChannel in
+                connectionChannel.eventLoop.makeCompletedFuture {
+                    try connectionChannel.pipeline.syncOperations.addHandler(waitForConnectionHandler)
+                }
+            }
+            .configureNWParameters { _ in
+                configuratorListenerCounter.withLockedValue { $0 += 1 }
+            }
+            .bind(host: "localhost", port: 0)
+            .get()
+
+        let connectionChannel: Channel = try await NIOTSConnectionBootstrap(group: group)
+            .configureNWParameters { _ in
+                configuratorConnectionCounter.withLockedValue { $0 += 1 }
+            }
+            .connect(to: listenerChannel.localAddress!)
+            .get()
+
+        // Wait for the server to activate the connection channel to the client.
+        try await waitForConnectionHandler.connectionPromise.futureResult.get()
+
+        try await listenerChannel.close().get()
+        try await connectionChannel.close().get()
+
+        XCTAssertEqual(2, configuratorListenerCounter.withLockedValue { $0 })
+        XCTAssertEqual(1, configuratorConnectionCounter.withLockedValue { $0 })
+
+        try await group.shutdownGracefully()
```

Comment:
This should ideally be done in a `defer` when created otherwise we might not shut it down if the test throws